api/v1beta1/azuremachinetemplate_webhook.go

```diff
@@ -35,41 +39,54 @@ const (
 // SetupWebhookWithManager sets up and registers the webhook with the manager.
 func (r *AzureMachineTemplate) SetupWebhookWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewWebhookManagedBy(mgr).
-		For(r).
+		For(&AzureMachineTemplate{}).
+		WithValidator(r).
+		WithDefaulter(r).
 		Complete()
 }
 
 // +kubebuilder:webhook:verbs=create;update,path=/mutate-infrastructure-cluster-x-k8s-io-v1beta1-azuremachinetemplate,mutating=true,failurePolicy=fail,matchPolicy=Equivalent,groups=infrastructure.cluster.x-k8s.io,resources=azuremachinetemplates,versions=v1beta1,name=default.azuremachinetemplate.infrastructure.cluster.x-k8s.io,sideEffects=None,admissionReviewVersions=v1;v1beta1
 // +kubebuilder:webhook:verbs=create;update,path=/validate-infrastructure-cluster-x-k8s-io-v1beta1-azuremachinetemplate,mutating=false,failurePolicy=fail,matchPolicy=Equivalent,groups=infrastructure.cluster.x-k8s.io,resources=azuremachinetemplates,versions=v1beta1,name=validation.azuremachinetemplate.infrastructure.cluster.x-k8s.io,sideEffects=None,admissionReviewVersions=v1;v1beta1
 
-var _ webhook.Defaulter = &AzureMachineTemplate{}
-var _ webhook.Validator = &AzureMachineTemplate{}
+var _ webhook.CustomDefaulter = &AzureMachineTemplate{}
+var _ webhook.CustomValidator = &AzureMachineTemplate{}
 
-// ValidateCreate implements webhook.Validator so a webhook will be registered for the type.
-func (r *AzureMachineTemplate) ValidateCreate() error {
-	spec := r.Spec.Template.Spec
+// ValidateCreate implements webhook.CustomValidator so a webhook will be registered for the type.
+func (r *AzureMachineTemplate) ValidateCreate(ctx context.Context, obj runtime.Object) error {
+	t, ok := obj.(*AzureMachineTemplate)
+	if !ok {
+		return apierrors.NewBadRequest(fmt.Sprintf("expected a AzureMachineTemplate but got a %T", obj))
+	}
```

Comment:
```suggestion
	t := obj.(*AzureMachineTemplate)
```
If you want to, given that in ValidateUpdate it was already done like this.
Theoretically this could panic, but the webhook is never called with another type

Reply:
(same in Default)

Reply:
I can see this throwing off a linter as well so probably safest to keep the long version

Reply:
The one in l.80 was already there before :)

Reply:
But up to you anyway. But might be good to do it consistently

Reply:
cool, let's keep this consistent with existing code and do the simple type change

@mboersma curious why we don't lint on this? seems the codebase would benefit from ensuring that we don't have these unchecked runtime type coercion fireworks...

Reply:
One more in Default() :)

Reply:
done